crates/ty_python_semantic/src/types/generics.rs

```diff
@@ -163,8 +163,32 @@ impl<'db> GenericContext<'db> {
         db: &'db dyn Db,
         types: Box<[Type<'db>]>,
     ) -> Specialization<'db> {
+        // Typevars can have other typevars as their default values, e.g.
+        //
+        // ```py
+        // class C[T, U = T]: ...
+        // ```
+        //
+        // If there is a mapping for `T`, we want to map `U` to that type, not to `T`. To handle
+        // this, we repeatedly apply the specialization to itself, until we reach a fixed point.
+        //
+        // This loop will always terminate, since it's not possible to create cyclic typevar
+        // references:
+        //
+        // ```py
+        // # error: [unresolved-reference] for use of `U`
+        // class C[T = U, U = T]: ...
```

Comment:
What about with generic typevars?

Reply:
This is now moot, but for posterity, the [typing spec requires](https://typing.python.org/en/latest/spec/generics.html#using-another-type-parameter-as-default) defaults to only refer to typevars that appear earlier in the generic context.